compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/inference/FirBuilderInferenceSession.kt

```diff
@@ -216,3 +226,26 @@ class ConeComposedSubstitutor(val left: ConeSubstitutor, val right: ConeSubstitu
         return left.substituteOrNull(rightSubstitution ?: type)
     }
 }
+
+class FirStubTypeTransformer(
+    override val session: FirSession,
+    private val substitutor: ConeSubstitutor
+) : FirAbstractTreeTransformer<Void?>(phase = FirResolvePhase.IMPLICIT_TYPES_BODY_RESOLVE) {
```

Comment:
It's better to use `Nothing?` or `Unit` (I prefer the first one) in such cases

Reply:
Thanks, I prefer `Nothing?` too.